### PR TITLE
FIX: `overlay remove` flags

### PR DESCRIPTION
**Pull request by amtoine**
This PR:
1. changes the mentions to the `--keep` flag of the `overlay remove` command, which does not exist, to `--keep-custom`, which does the same thing
2. adds a small paragraph about the `--keep-env` flag of the `overlay remove` command

NOTES:
- i've changed all mentions to `--keep` using `sed` and `ls **/*`, so there shouldn't be any `--keep` left :ok_hand: 
- i'm not able to write a chinese version of my new paragraph inside `zh-CH/book/overlays.md` :confused:

### CODE REVIEW

File: book/overlays.md [hunk 1]
```diff
@@ -186,13 +186,31 @@ Sometimes, you might want to remove an overlay, but keep all the custom definiti
...
+(spam)> overlay remove --keep-custom spam
 
 (zero)> eggs
 eggs
 ```
 
-The `--keep` flag does exactly that.
+The `--keep-custom` flag does exactly that.
+
+One can also keep a set of environment variables that were defined inside an overlay, but remove the rest:
```
Comment (kubouch) [inline book/overlays.md:197]: I'd rephrase it to something like "One can also keep a list of environment variables that were defined inside an overlay, but remove the rest, using the `--keep-env` flag:" (note the set->list change as well)

File: book/overlays.md [hunk 2]
```diff
@@ -186,13 +186,31 @@ Sometimes, you might want to remove an overlay, but keep all the custom definiti
...
+(spam)> overlay remove spam --keep-env [FOO]
+
+(zero)> foo
+Error: Can't run executable...
+
+(zero)> $env.FOO
+foo
+```
+
+The `--keep-env` flag does exactly that.
```
Comment (kubouch) [inline book/overlays.md:213]: ... and I'd remove this line

File: book/overlays.md (hunk 1 above)
Comment (amtoine) [reply book/overlays.md:197]: yep, makes sense :ok_hand:

File: book/overlays.md (hunk 2 above)
Comment (amtoine) [reply book/overlays.md:213]: i kept the same format as the `--keep-custom` flag, but yeah sure :+1:

Comment (amtoine) [reply book/overlays.md]: should be addressed in [4df4f74](https://github.com/nushell/nushell.github.io/pull/565/commits/4df4f74eb88795e9126ae8324bbc31e676b75252) :yum:

File: book/overlays.md (hunk 1 above)
Comment (amtoine) [reply book/overlays.md]: should be addressed in [320ac57](https://github.com/nushell/nushell.github.io/pull/565/commits/320ac576c09e527acaae2a8caac9c2cc7ad4e72c) :yum:

Comment (kubouch) [reply book/overlays.md]: I meant the `--keep-env` flag should still be mentioned in the text. It just felt like the "
The `--keep-env` flag does exactly that." line later on was a bit out of place.

Comment (amtoine) [reply book/overlays.md]: ooooh sorry, i was too much focused on the `set->list` change :eyes:
got it now :wink:

Comment (amtoine) [reply book/overlays.md]: there should be the changes :relieved:
[1a2a41](https://github.com/nushell/nushell.github.io/pull/565/commits/1a2a41c669d34f67d86e75adbcc7a4f069ecfdfd)

**Comment by kubouch** [comment]
Perfect, thanks!

**Comment by amtoine** [comment]
cooool :yum:
